pkg/clusters/types/eks/aws-operations/eks.go

```diff
@@ -0,0 +1,479 @@
+		DesiredCapacity: aws.Int(1),
+		MinSize:         aws.Int(1),
+		MaxSize:         aws.Int(1),
+	}
+
+	nodeKeyName := os.Getenv(envKeyNodeSSHKeyName)
+	if nodeKeyName != "" {
+		ng.SSH.Allow = aws.Bool(true)
+		ng.SSH.PublicKeyName = aws.String(nodeKeyName)
+	}
+
+	return clusterCfg
+}
+
+func waitForClusterActive(ctx context.Context, eksClient *eks.Client, clusterName string) (*types.Cluster, error) {
+	childCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+	ticker := time.NewTicker(10 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-childCtx.Done():
+			return nil, childCtx.Err()
+		case <-ticker.C:
+			describeInput := &eks.DescribeClusterInput{
+				Name: &clusterName,
+			}
+			resp, err := eksClient.DescribeCluster(ctx, describeInput)
+			if err != nil {
+				return nil, errors.Wrap(err, fmt.Sprintf("failed to describe EKS cluster %s", clusterName))
+			}
+
+			status := resp.Cluster.Status
+			if status == types.ClusterStatusActive {
+				return resp.Cluster, nil
+			}
+		}
+	}
+}
+
+func authorizeNodeGroup(clientSet kubernetes.Interface, nodeRoleArn string) error {
+	acm, err := authconfigmap.NewFromClientSet(clientSet)
+	if err != nil {
+		return err
+	}
+
+	nodeGroupRoles := authconfigmap.RoleNodeGroupGroups
+
+	identity, err := eksiam.NewIdentity(nodeRoleArn, authconfigmap.RoleNodeGroupUsername, nodeGroupRoles)
+	if err != nil {
+		return err
+	}
+
+	if err := acm.AddIdentity(identity); err != nil {
+		return errors.Wrap(err, "adding nodegroup to auth ConfigMap")
+	}
+	if err := acm.Save(); err != nil {
+		return errors.Wrap(err, "saving auth ConfigMap")
+	}
+	return nil
+}
+
+func createNodeGroup(ctx context.Context, eksClient *eks.Client, ec2Client *ec2.Client, clusterCfg *eksctlapi.ClusterConfig) error {
+	nodeGroup := clusterCfg.NodeGroups[0]
+	launchTemplateId, err := createNodeLaunchTemplate(ctx, ec2Client, clusterCfg)
+	if err != nil {
+		return errors.Wrap(err, "failed to create launch template")
+	}
+
+	input := &eks.CreateNodegroupInput{
+		ClusterName:   aws.String(clusterCfg.Metadata.Name),
+		NodegroupName: aws.String(nodeGroup.Name),
+		NodeRole:      aws.String(nodeGroup.IAM.InstanceRoleARN),
+		Subnets:       nodeGroup.Subnets,
+		ScalingConfig: &types.NodegroupScalingConfig{
+			MinSize:     aws.Int32(int32(aws.ToInt(nodeGroup.MinSize))),
+			MaxSize:     aws.Int32(int32(aws.ToInt(nodeGroup.MaxSize))),
+			DesiredSize: aws.Int32(int32(aws.ToInt(nodeGroup.DesiredCapacity))),
+		},
+		LaunchTemplate: &types.LaunchTemplateSpecification{
+			Id: aws.String(launchTemplateId),
+		},
+	}
+
+	_, err = eksClient.CreateNodegroup(ctx, input)
+	if err != nil {
+		return err
+	}
+
+	return waitForNodeGroupReady(ctx, eksClient, clusterCfg.Metadata.Name, nodeGroup.Name)
+}
+
+func waitForNodeGroupReady(ctx context.Context, eksClient *eks.Client, clusterName, nodeGroupName string) error {
+	childCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
```

Comment:
Ditto here for wait time and wait tick.

Reply:
Waiting for a node group to become ready can take about 5 minutes. 
I'll change to 5 seconds here.